sycl/include/sycl/ext/oneapi/matrix/static-query-use.hpp

```diff
@@ -482,6 +482,379 @@ struct matrix_params<
   template <typename Group>
   using joint_matrix_d = joint_matrix<Group, Td, use::accumulator, M, N>;
 };
+
+//////////////////////////////////////////////
+/// AMD Matrix Cores - GFX90A architecture ///
+//////////////////////////////////////////////
+
+template <typename Ta, typename Tc>
+constexpr bool is_combination_valid_amd_gfx90a(size_t sM, size_t sN,
+                                               size_t sK) {
+  if ((std::is_same_v<Ta, half> && std::is_same_v<Tc, float> &&
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t> &&
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, unsigned short> && std::is_same_v<Tc, float> &&
+       ((sM == 32 && sN == 32 && sK == 8) ||
+        (sM == 16 && sN == 16 && sK == 16))) ||
+      (std::is_same_v<Ta, double> && std::is_same_v<Tc, double> &&
+       (sM == 16 && sN == 16 && sK == 4)))
+    return true;
+  else
+    return false;
+}
+
```

Comment:
I've just tried it this way but the code now looks quite unreadable due to the one more OR in that case:
This would take shape of: the above conditions ORed with the extra case for `double`: 
```c++
  if ((((sM == 32 && sN == 32 && sK == 8) ||
        (sM == 16 && sN == 16 && sK == 16)) &&
           (std::is_same_v<Ta, half> && std::is_same_v<Tc, float>) ||
       (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t>) ||
       (std::is_same_v<Ta, bfloat16> && std::is_same_v<Tc, float>)) ||
      ((sM == 16 && sN == 16 && sK == 4) &&
       (std::is_same_v<Ta, double> && std::is_same_v<Tc, double>)))
```
btw, this is already after applying clang-format. I think for the sake of readability this should be left as is.